tests/test_atoms_state.py

```diff
@@ -0,0 +1,386 @@
+        [
+            ('l', 'number', 0, 'symbol', 's'),
+            ('l', 'number', 1, 'symbol', 'p'),
+            ('l', 'number', 2, 'symbol', 'd'),
+            ('l', 'number', 3, 'symbol', 'f'),
+            ('l', 'number', 4, 'symbol', None),
+            ('ml', 'number', -1, 'symbol', 'x'),
+            ('ml', 'number', 0, 'symbol', 'z'),
+            ('ml', 'number', 1, 'symbol', 'y'),
+            ('ml', 'number', -2, 'symbol', None),
+            ('ms', 'number', -0.5, 'symbol', 'down'),
+            ('ms', 'number', 0.5, 'symbol', 'up'),
+            ('ms', 'number', -0.75, 'symbol', None),
+            ('no_attribute', 'number', None, 'symbol', None),
+        ],
+    )
+    def test_number_and_symbol(
+        self,
+        orbital_state,
+        quantum_name,
+        quantum_type,
+        value,
+        countertype,
+        expected_result,
+    ):
+        """
+        Test the number and symbol resolution for each of the quantum numbers defined in the parametrization.
+        """
+        # Adding quantum numbers to the `OrbitalsState` section
+        self.add_quantum_numbers(orbital_state, quantum_name, quantum_type, value)
+
+        # Making sure that the `quantum_type` is assigned
+        resolved_type = orbital_state.resolve_number_and_symbol(
+            quantum_name, quantum_type, self.logger
+        )
+        assert resolved_type == value
+
+        # Resolving if the counter-type is assigned
+        resolved_countertype = orbital_state.resolve_number_and_symbol(
+            quantum_name, countertype, self.logger
+        )
+        assert resolved_countertype == expected_result
+
+    @pytest.mark.parametrize(
+        'l_quantum_number, ml_quantum_number, j_quantum_number, mj_quantum_number, ms_quantum_number, degeneracy',
+        [
+            (1, None, None, None, 0.5, 3),
+            (1, None, None, None, None, 6),
+            (1, -1, None, None, 0.5, 1),
+            (1, -1, None, None, None, 2),
+            # ! these J and MJ tests are unphysical; check what is going on
+            (1, -1, [1 / 2, 3 / 2], None, None, 6),
+            (1, -1, [1 / 2, 3 / 2], [-3 / 2, 1 / 2, 1 / 2, 3 / 2], None, 2),
+        ],
+    )
+    def test_degeneracy(
+        self,
+        orbital_state,
+        l_quantum_number,
+        ml_quantum_number,
+        j_quantum_number,
+        mj_quantum_number,
+        ms_quantum_number,
+        degeneracy,
+    ):
+        """
+        Test the degeneracy of each of orbital states defined in the parametrization.
+        """
+        self.add_state(
+            orbital_state,
+            l_quantum_number,
+            ml_quantum_number,
+            ms_quantum_number,
+            j_quantum_number,
+            mj_quantum_number,
+        )
+        resolved_degeneracy = orbital_state.resolve_degeneracy()
+        assert resolved_degeneracy == degeneracy
+
+    def test_normalize(self, orbital_state):
```

Comment:
see other comment

Reply:
(just resolving in favor of the other comment)